---
title: C6255
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C6255
helpviewer_keywords:
- C6255
ms.assetid: bb6430b2-782a-4410-a8e1-609df06007de
author: mikeblome
ms.author: mblome
manager: markl
ms.workload:
- multiple
ms.openlocfilehash: 8207a376b314ef6f13c2e1ff4af0dcd5991c668c
ms.sourcegitcommit: 5f6ad1cefbcd3d531ce587ad30e684684f4c4d44
ms.translationtype: MT
ms.contentlocale: pt-BR
ms.lasthandoff: 10/22/2019
ms.locfileid: "72746638"
---
# <a name="c6255"></a>C6255
aviso C6255: _alloca indica falha ao gerar uma exceção de estouro de pilha. Considere usar _malloca em vez disso

 Esse aviso indica que uma chamada para `_alloca` foi detectada fora do tratamento de exceção local. `_alloca` sempre deve ser chamado de dentro do intervalo protegido de um manipulador de exceção, pois ele pode gerar uma exceção de estouro de pilha em caso de falha. Se possível, em vez de usar `_alloca`, considere o uso de `_malloca` que é uma versão mais segura do `_alloca`.

## <a name="example"></a>Exemplo
 O código a seguir gera esse aviso porque `_alloca` pode gerar a exceção:

```
#include <windows.h>

void f( )
{
  void *p = _alloca(10);
  // code ...
}
```

 Para corrigir esse aviso, use `_malloca` e adicione o manipulador de exceção, conforme mostrado no código a seguir:

```
#include <windows.h>
#include <malloc.h>

void f( )
{
  void *p;
  int errcode;
  __try
  {
    p = _malloca(10);
    // code...
    _freea(p);
  }
  __except( (GetExceptionCode() == STATUS_STACK_OVERFLOW ) ?
              EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH )
  {
    errcode = _resetstkoflw();
    // code ...
  }
}
```

## <a name="see-also"></a>Consulte também
 [_malloca](/cpp/c-runtime-library/reference/malloca)
